### PR TITLE
Update the pin on `prefect` version

### PR DESCRIPTION
**Pull request by urimandujano**
As part of our work adding support for Pydantic 2, we removed its pin in 
`prefect`'s `requirements.txt`. This means that it's possible to have 
`prefect`, `pydantic>=2`, and any version of this collection installed. But, 
the collection(s) only support `pydantic>=2` in their latest versions. This 
PR adds a pin on the collection's `requirements.txt` to make sure that it is 
only installed with the correct version `prefect` that supports `pydantic>=2`

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-prefect>=2.10.11
+prefect >=2.13.5
```
Comment (chrisguidry) [inline requirements.txt:1]: Nit: the other lines don't have the space before the `>=`? :stuck_out_tongue:

Comment (urimandujano) [reply requirements.txt:1]: Good point, right before the flood gates opened too!

Comment (chrisguidry) [reply requirements.txt:1]: :D I figured I should catch you quick

**Review by chrisguidry — APPROVED** [review]
:shipit:

**Pull request merged**